pkg/trtl/trtl_test.go

```diff
@@ -132,6 +148,17 @@ func (s *trtlTestSuite) SetupSuite() {
 	os.Setenv("TRTL_REPLICA_REGION", "minneapolis")
 	s.conf, err = config.New()
 	require.NoError(err)
+
+	// Create the trtl server
+	s.trtl, err = trtl.New(s.conf)
+	require.NoError(err)
+
+	// Create a bufcon listener so that there are no actual network requests
+	s.conn = bufconn.Listen(bufSize)
+
+	// Run the test server without signals, background routines or maintenance mode checks
+	// TODO: do we need to check if there was an error when starting run?
+	go s.trtl.Run(s.conn)
```

Comment:
This refactoring allows us to run the trtl server without dialing a real network address and taking a port during tests. The buffconn is also faster to clean up and allows in-memory tests.